### ISSUE TITLE
osd is not coming up

### ISSUE THREAD

**Issue by vadlakiran** [rook/rook#13501]
Hi Team,
we are trying to install v1.8.1 we have 3 worker nodes and 2 osd came up, once of the osd is not coming up, its going prepared pod is in crashloopbackoff and below is the logs i can see specific worker node logs from operator pod logs. please help us to get it done this ?

`2024-01-04 06:13:46.615669 I | op-osd: OSD orchestration status for node worker6 is "failed"
2024-01-04 06:13:46.615740 E | op-osd: failed to provision OSD(s) on node worker6. &{OSDs:[] Status:failed PvcBackedOSD:false Message:failed to configure devices: failed to generate osd keyring: failed to get or create auth key for client.bootstrap-osd: failed get-or-create-key client.bootstrap-osd: exit status 1}
2024-01-04 06:13:46.636137 E | ceph-cluster-controller: failed to reconcile CephCluster "rook-ceph/rook-ceph". failed to reconcile cluster "rook-ceph": failed to configure local ceph cluster: failed to create cluster: failed to start ceph osds: 1 failures encountered while running osds on nodes in namespace "rook-ceph".
failed to provision OSD(s) on node worker6. &{OSDs:[] Status:failed PvcBackedOSD:false Message:failed to configure devices: failed to generate osd keyring: failed to get or create auth key for client.bootstrap-osd: failed get-or-create-key client.bootstrap-osd: exit status 1}
2024-01-04 06:13:46.636185 I | op-k8sutil: Reporting Event rook-ceph:rook-ceph Warning:ReconcileFailed:failed to reconcile cluster "rook-ceph": failed to configure local ceph cluster: failed to create cluster: failed to start ceph osds: 1 failures encountered while running osds on nodes in namespace "rook-ceph".
failed to provision OSD(s) on node worker6. &{OSDs:[] Status:failed PvcBackedOSD:false Message:failed to configure devices: failed to generate osd keyring: failed to get or create auth key for client.bootstrap-osd: failed get-or-create-key client.bootstrap-osd: exit status 1}
2024-01-04 06:13:46.642055 I | ceph-cluster-controller: reconciling ceph cluster in namespace "rook-ceph"
2024-01-04 06:13:46.647613 I | op-mon: parsing mon endpoints: c=10.233.5.31:6789,a=10.233.30.135:6789,b=10.233.15.151:6789
2024-01-04 06:13:46.679219 I | ceph-spec: detecting the ceph image version for image docker-registry.com:5000/ceph/ceph:v16.2.7...
2024-01-04 06:13:49.625044 I | ceph-spec: detected ceph image version: "16.2.7-0 pacific"
2024-01-04 06:13:49.625071 I | ceph-cluster-controller: validating ceph version from provided image
2024-01-04 06:13:49.631208 I | op-mon: parsing mon endpoints: c=10.233.5.31:6789,a=10.233.30.135:6789,b=10.233.15.151:6789
2024-01-04 06:13:49.633606 I | cephclient: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2024-01-04 06:13:49.633933 I | cephclient: generated admin config in /var/lib/rook/rook-ceph`

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

**Expected behavior:**

**How to reproduce it (minimal and precise):**

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary

**Logs to submit**:

* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

  To get logs, use `kubectl -n <namespace> logs <pod name>`
  When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
  Read [GitHub documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Cluster Status to submit**:

* Output of kubectl commands, if necessary

  To get the health of the cluster, use `kubectl rook-ceph health`
  To get the status of the cluster, use `kubectl rook-ceph ceph status`
  For more details, see the [Rook kubectl Plugin](https://rook.io/docs/rook/latest-release/Troubleshooting/kubectl-plugin)

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest-release/Troubleshooting/ceph-toolbox/#interactive-toolbox)):

**Comment by travisn** [comment]
This is the key failure:
```
2024-01-04 06:13:46.615740 E | op-osd: failed to provision OSD(s) on node worker6. &{OSDs:[] Status:failed PvcBackedOSD:false 
Message:failed to configure devices: failed to generate osd keyring: 
failed to get or create auth key for client.bootstrap-osd: 
failed get-or-create-key client.bootstrap-osd: exit status 1}
```
The same bootstrap-osd keyring is used across all the osds, so this is a strange failure since two of your OSDs did start successfully. Some possibilities:
- Does this repro? Or you have only seen it in this single cluster?
- Is it an option to clean the cluster and install again?
- Perhaps try `ceph auth rm client.bootstrap-osd` and then restart the operator to see if it regenerating the keyring will allow it to succeed.

**Comment by vadlakiran** [comment]
@travisn  yes we can repro, its in one cluster we have this issue
yes we have option to clean the cluster and install it again, but i tried multiple times but same issue we are getting
ceph auth rm client.bootstrap-osd  need to run from tools right, will it effect other osd's right because the worker6 osd is not yet all connecting

**Comment by subhamkrai** [comment]
@vadlakiran yes `ceph auth rm client.bootstrap-osd` this needs to run with tool and it will have some effect on other osd but since your cluster seems to new/getting-started you can try to see if this helps?

also could you share the osd-prepare logs of the osd with is not coming?

**Comment by vadlakiran** [comment]
@subhamkrai  still same issue below are latest preared logs

`2024-01-05 16:58:01.312016 D | exec: Running command: ceph-volume inventory --format json /dev/sdb
2024-01-05 16:58:04.206371 I | cephosd: device "sdb" is available.
2024-01-05 16:58:04.206399 I | cephosd: "sdb" found in the desired devices
2024-01-05 16:58:04.206404 I | cephosd: device "sdb" is selected by the device filter/name "sdb"
2024-01-05 16:58:04.214047 I | cephosd: configuring osd devices: {"Entries":{"sdb":{"Data":-1,"Metadata":null,"Config":{"Name":"sdb","OSDsPerDevice":1,"MetadataDevice":"","DatabaseSizeMB":0,"DeviceClass":"hdd","InitialWeight":"","IsFilter":false,"IsDevicePathFilter":false},"PersistentDevicePaths":["/dev/disk/by-path/pci-0000:00:10.0-scsi-0:0:1:0"],"DeviceInfo":{"name":"sdb","parent":"","hasChildren":false,"devLinks":"/dev/disk/by-path/pci-0000:00:10.0-scsi-0:0:1:0","size":53687091200,"uuid":"a0da62af-0146-4bbb-a02f-68f4bc41feba","serial":"","type":"disk","rotational":true,"readOnly":false,"Partitions":null,"filesystem":"","vendor":"VMware","model":"Virtual_disk","wwn":"","wwnVendorExtension":"","empty":false,"real-path":"/dev/sdb","kernel-name":"sdb"}}}}
2024-01-05 16:58:04.219491 I | cephclient: getting or creating ceph auth key "client.bootstrap-osd"
2024-01-05 16:58:04.219565 D | exec: Running command: ceph auth get-or-create-key client.bootstrap-osd mon allow profile bootstrap-osd --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --name=client.admin --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json
`

**Comment by travisn** [comment]
If it's consistent, perhaps there is a networking issue from this node. The ceph command is not able to communicate from the osd prepare job on that node to the ceph mons.

**Comment by vadlakiran** [comment]
@travisn @subhamkrai Thank you, i have re-installed and reproduced the issue and i executed below command and its cleaned then i reinstalled it again, its resolved the issue.

ceph auth rm client.bootstrap-osd

**Comment by travisn** [comment]
Good to hear it's resolved, so i'll close it.
